submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/ProjectResource.kt

```diff
@@ -18,21 +22,33 @@ import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.ResponseBody
 import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.multipart.MultipartFile
+import springfox.documentation.annotations.ApiIgnore
 
 @RestController
 @RequestMapping("/projects")
 @PreAuthorize("isAuthenticated()")
+@Api(tags = ["Projects"])
 class ProjectResource(private val projectService: ProjectService, private val projectWebHandler: ProjectWebHandler) {
     @GetMapping
     @ResponseBody
+    @ApiOperation("Get the list of available projects for the current user")
+    @ApiImplicitParam(name = "X-Session-Token", value = "The authentication token", required = true)
     fun getUserProjects(
-        @AuthenticationPrincipal user: SecurityUser
+        @ApiIgnore
+        @AuthenticationPrincipal
+        user: SecurityUser
     ): List<Project> = projectService.getAllowedProjects(user, AccessType.ATTACH)
 
     @PostMapping(headers = ["${HttpHeaders.CONTENT_TYPE}=$MULTIPART_FORM_DATA"])
     @ResponseBody
+    @ApiOperation("Register a new project")
+    @ApiImplicitParam(name = "X-Session-Token", value = "The authentication token", required = true)
     fun submit(
-        @AuthenticationPrincipal user: SecurityUser,
+        @ApiIgnore
+        @AuthenticationPrincipal
+        user: SecurityUser,
```

Comment:
I am wondering if ` .ignoredParameterTypes(AuthenticationPrincipal.class)` does not work?

Reply:
You're right, nice catch!